modin/backends/base/query_compiler.py

```diff
@@ -1740,218 +3577,1371 @@ def set_index_names(self, names, axis=0):
+    # should be removed.
+    @_doc_resample_fillna(method="forward-fill", link="ffill")
+    def resample_ffill(self, resample_args, limit):
+        return ResampleDefault.register(pandas.core.resample.Resampler.ffill)(
+            self, resample_args, limit
+        )
+
+    # FIXME: we should combine all method all resample fillna methods into `resample_fillna`
+    @_doc_resample_fillna(method="specified", link="fillna", params="method : str")
+    def resample_fillna(self, resample_args, method, limit):
+        return ResampleDefault.register(pandas.core.resample.Resampler.fillna)(
+            self, resample_args, method, limit
+        )
+
+    @_doc_resample_reduction(
+        result="first element", link="first", params="_method : str"
+    )
+    def resample_first(self, resample_args, _method, *args, **kwargs):
+        return ResampleDefault.register(pandas.core.resample.Resampler.first)(
+            self, resample_args, _method, *args, **kwargs
+        )
+
+    # FIXME: high-level objects leaks to the query compiler.
+    def resample_get_group(self, resample_args, name, obj):
+        """
+        Resample time-series data and get the specified group.
+
+        Group data into intervals by time-series row/column with
+        a specified frequency and get the values of the specified group.
+
+        Parameters
+        ----------
+        resample_args : list
+            Resample parameters in the format of ``modin.pandas.DataFrame.resample`` signature.
+        name : object
+        obj : modin.pandas.DataFrame, optional
+
+        Returns
+        -------
+        BaseQueryCompiler
+            New QueryCompiler containing the values from the specified group.
+        """
+        return ResampleDefault.register(pandas.core.resample.Resampler.get_group)(
+            self, resample_args, name, obj
+        )
+
+    @_doc_resample_fillna(
+        method="specified interpolation",
+        link="interpolate",
+        params="""method : str
+        axis : {0, 1}
+        limit : int
+        inplace : {False}
+            This parameter serves the compatibility purpose. Always have to be False.
+        limit_direction : {"forward", "backward", "both"}
+        limit_area : {None, "inside", "outside"}
+        downcast : str, optional
+        **kwargs : dict""",
+        overwrite_template_params=True,
+    )
+    def resample_interpolate(
+        self,
+        resample_args,
+        method,
+        axis,
+        limit,
+        inplace,
+        limit_direction,
+        limit_area,
+        downcast,
+        **kwargs,
+    ):
+        return ResampleDefault.register(pandas.core.resample.Resampler.interpolate)(
+            self,
+            resample_args,
+            method,
+            axis,
+            limit,
+            inplace,
+            limit_direction,
+            limit_area,
+            downcast,
+            **kwargs,
+        )
```

Comment:
this would certainly benefit in readability from being replaced by
```python
resample_interpolate = _doc_resample_fillna(......)(ResampleDefault.register(pandas.core.resample.Resampler.interpolate))
```

Reply:
but then we lose the signature of this function, the function would just take `*args` and `**kwargs` and it would be completely unclear what the actual parameters of this method. Function without solid signature more likely to get unsynced with its docs.